### ISSUE TITLE
Build failure (GCC 7.3.0)

### ISSUE THREAD

**Issue by kodo-pp** [Andrewerr/cddos#15]
Release build fails when compiling with GCC 7.3.0.
I didn't try debug build, but I guess the result will be the same

[build log](https://gist.github.com/kodo-pp/8e53289790cf3f521f9901e0b672291d)

Possible reason: some includes are missing

UPD: Tried [these](https://gist.github.com/kodo-pp/bbdc286c9db915fa190b9dcf6a5446a0) changes, but something in memcrashed.c is wrong

**Comment by 0xf104a** [comment]
Current repo version isn't ready for building.I haven't written cross platform low-level raw sockets.So memcrashed.c is now compatible with Mac OS X only.Try building version from here https://github.com/Andrewerr/ddos/archive/1.1R3.tar.gz 
**P.S.:** In 1.1R3 types like uint8_t also wouldn't work with gcc 7.3.0.You'll still need to do the same changes you done to make it to compile

**Comment by kodo-pp** [comment]
OK, I've just made it compile on my PC

This patch should work (apply it to commit 67682aa; also there is no memcrashed support (its functions are just replaced with stubs))
```
diff --git a/build.sh b/build.sh
index 725a3e2..5326fdb 100755
--- a/build.sh
+++ b/build.sh
@@ -58,7 +58,7 @@ fi
 if [ $1 = "release" ]
   then
    echo "Building release"
-   gcc ping.c util.c message.c socket.c ddos.c main.c -lm -Ofast -Os -lpthread -o ddos
+   gcc memcrashed_dummy.c ping.c util.c message.c socket.c ddos.c main.c -lm -Ofast -Os -lpthread -o ddos
    exit
 fi
 if [ $1 = "update" ]
diff --git a/memcrashed_dummy.c b/memcrashed_dummy.c
new file mode 100644
index 0000000..d57adfe
--- /dev/null
+++ b/memcrashed_dummy.c
@@ -0,0 +1,18 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include "memcrashed.h"
+
+void __memcrashed_abort(void) {
+    fprintf(stderr, "Memcrashed is not included in the build. Aborting\n");
+    abort();
+}
+
+void memcrashed_status(void) {
+    __memcrashed_abort();
+}
+void memcrashed_ddos(_dos_param* p) {
+    __memcrashed_abort();
+}
+void memcrashed_init(const char* hostfile) {
+    __memcrashed_abort();
+}
diff --git a/ping.c b/ping.c
index 9aa684a..2305a57 100644
--- a/ping.c
+++ b/ping.c
@@ -17,8 +17,8 @@
 //Code of this module is based on https://github.com/hayderimran7/advanced-socket-programming/blob/master/ping.c
 #ifdef __ANDROID__
 #warning Non-rooted android devices does not support ping module
-#define errno_t int
 #endif
+#define errno_t int
 
 #define DEBUG
 
diff --git a/types.h b/types.h
index b60201d..e69de29 100644
--- a/types.h
+++ b/types.h
@@ -1,53 +0,0 @@
-//
-//  types.h
-//  cdos
-//
-//  Created by System Administrator on 12.05.2018.
-//  Copyright © 2018 Andre Zay. All rights reserved.
-//
-
-#ifndef types_h
-#define types_h
-/*Crosspaltform/crosscompiler types definitions*/
-
-#ifndef _INT64_T
-#define _INT64_T
-typedef long long int64_t;
-#endif
-
-#ifndef _INT32_T
-#define _INT32_T
-typedef int int32_t;
-#endif
-
-#ifndef _INT16_T
-#define _INT16_T
-typedef short int16_t;
-#endif
-
-#ifndef _INT8_T
-#define _INT8_T
-typedef char int8_t;
-#endif
-
-#ifndef _UINT64_T
-#define _UINT64_T
-typedef unsigned long long uint64_t;
-#endif
-
-#ifndef _UINT32_T
-#define _UINT32_T
-typedef unsigned int uint32_t;
-#endif
-
-#ifndef _UINT16_T
-#define _UINT16_T
-typedef unsigned short uint16_t;
-#endif
-
-#ifndef _UINT8_T
-#define _UINT8_T
-typedef unsigned char uint8_t;
-#endif
-
-#endif /* types_h */
```

**Comment by 0xf104a** [comment]
Thank you a lot for your participation and collaboration.I've fixed this issue in my latest commit (a43a5bb)
